### ISSUE TITLE
Allow many_single_char_names still produces warnings

### ISSUE THREAD

**Issue by bluss** [rust-lang/rust-clippy#5521]
#![allow(clippy::many_single_char_names)] does not seem to suppress the lint, as it should,

It produces warnings like

```rust
warning: 6 bindings with single-character names in scope
   --> src\dimension\mod.rs:422:17
    |
422 | fn extended_gcd(a: isize, b: isize) -> (isize, (isize, isize)) {
    |                 ^         ^
...

```

Lints allowed in [the code](https://github.com/rust-ndarray/ndarray/blob/78bd1d84b9baf26c82c299e3bada43f72aa04614/src/lib.rs):

```rust
#![allow(
    clippy::many_single_char_names,
    clippy::deref_addrof,
    clippy::unreadable_literal,
    clippy::many_single_char_names
)]
```

Reproduced at ndarray commit 78bd1d84b9baf26c82c299e3bada43f72aa04614

Clippy version clippy 0.0.212 (6651c1b9 2020-04-15)

CI log with failure (clippy at the end) https://travis-ci.org/github/rust-ndarray/ndarray/jobs/678756560

Previous similar issue #3198

**Comment by flip1995** [comment]
This will be fixed with https://github.com/rust-lang/rust-clippy/pull/5518

**Comment by bluss** [comment]
nice!
